tests/bluetooth/bsim_bt/bsim_test_l2cap/src/main_l2cap_ecred.c

```diff
@@ -0,0 +1,618 @@
+/* main_l2cap_ecred.c - Application main entry point */
+
+/*
+ * Copyright (c) 2022 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "common.h"
+
+#define LOG_MODULE_NAME main_l2cap_ecred
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, LOG_LEVEL_DBG);
+
+extern enum bst_result_t bst_result;
+
+static struct bt_conn *default_conn;
+
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+};
+
+#define DATA_MTU CONFIG_BT_L2CAP_TX_MTU
+#define DATA_MPS 65
+#define DATA_BUF_SIZE BT_L2CAP_SDU_BUF_SIZE(DATA_MTU)
+#define L2CAP_CHANNELS 2
+#define SERVERS 1
+#define SDU_SEND_COUNT 200
+#define ECRED_CHAN_MAX 5
+#define LONG_MSG (DATA_MTU - 500)
+#define SHORT_MSG (DATA_MPS - 2)
+#define LONG_MSG_CHAN_IDX 0
+#define SHORT_MSG_CHAN_IDX 1
+
+NET_BUF_POOL_FIXED_DEFINE(rx_data_pool, L2CAP_CHANNELS, BT_L2CAP_BUF_SIZE(DATA_BUF_SIZE), 8, NULL);
+NET_BUF_POOL_FIXED_DEFINE(tx_data_pool_0, 1, BT_L2CAP_BUF_SIZE(DATA_MTU), 8, NULL);
+NET_BUF_POOL_FIXED_DEFINE(tx_data_pool_1, 1, BT_L2CAP_BUF_SIZE(DATA_MTU), 8, NULL);
+
+static struct bt_l2cap_server servers[SERVERS];
+void send_sdu_chan_worker(struct k_work *item);
+struct channel {
+	uint8_t chan_id; /* Internal number that identifies L2CAP channel. */
+	struct bt_l2cap_le_chan le;
+	bool in_use;
+	int sdus_received;
+	int bytes_to_send;
+	int iteration;
```

Comment:
Should/could be unsigned values

```suggestion
	size_t sdus_received;
	size_t bytes_to_send;
	uint8_t iteration;
```